src/braket/aws/aws_device.py

```diff
@@ -389,32 +390,48 @@ def get_devices(
         Returns:
             List[AwsDevice]: list of AWS devices
         """
-        order_by_list = ["arn", "name", "type", "provider_name", "status"]
-        if order_by not in order_by_list:
-            raise ValueError(f"order_by '{order_by}' must be in {order_by_list}")
-        device_regions_set = AwsDevice._get_devices_regions_set(arns, provider_names, types)
-        results = []
+        if order_by not in AwsDevice._GET_DEVICES_ORDER_BY_KEYS:
+            raise ValueError(
+                f"order_by '{order_by}' must be in {AwsDevice._GET_DEVICES_ORDER_BY_KEYS}"
+            )
+        aws_session = aws_session if aws_session else AwsSession()
+        types = set(types) if types else {AwsDeviceType.QPU, AwsDeviceType.SIMULATOR}
+        device_map = {}
+        device_regions_set = AwsDevice._get_devices_regions_set(arns, provider_names)
         for region in device_regions_set:
-            aws_session = AwsDevice._copy_aws_session(aws_session, [region])
-            results.extend(
-                aws_session.search_devices(
+            session_for_region = AwsDevice._copy_aws_session(aws_session, [region])
+            # Require simulators to be instantiated
+            # in the same region as the AWS session
```

Comment:
Is there an assumption here that the QPUs will always be restricted to a single region ?

Reply:
For the time being, yeah; the behavior of this method for QPUs is effectively unmodified. If a QPU is available in multiple regions, it will be instantiated in the first region it's available in alphabetically.